packages/shared/src/components/buttons/ButtonV2.tsx

```diff
@@ -0,0 +1,256 @@
+import React, {
+  forwardRef,
+  HTMLAttributes,
+  ReactElement,
+  ReactNode,
+  Ref,
+} from 'react';
+import classNames from 'classnames';
+import { IconProps, IconSize } from '../Icon';
+import { Loader } from '../Loader';
+import { combinedClicks } from '../../lib/click';
+
+import { ColorName as ButtonColor } from '../../styles/colors';
+
+export enum ButtonSize {
```

Comment:
Maybe we can move some of the function definitions to a `/buttons/common` file just to less bloat this page?

Reply:
Moved most of constants and the hook into the `common.ts` file.